django/thunderstore/frontend/api/experimental/views/community_package_list.py

```diff
@@ -0,0 +1,253 @@
+from django.db.models import Count, Prefetch, Q, QuerySet, Sum
+from django.http import HttpRequest, HttpResponse
+from drf_yasg.utils import swagger_auto_schema
+from rest_framework import status
+from rest_framework.generics import get_object_or_404
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from thunderstore.cache.cache import CacheBustCondition
+from thunderstore.cache.pagination import CachedPaginator
+from thunderstore.community.models import (
+    Community,
+    PackageListingReviewStatus,
+    PackageListingSection,
+)
+from thunderstore.frontend.api.experimental.serializers.views import (
+    CommunityPackageListSerializer,
+    PackageCategorySerializer,
+)
+from thunderstore.repository.models import Package
+
+
+class CommunityPackageListApiView(APIView):
+    """
+    Return paginated list of community's packages.
+    """
+
+    permission_classes = []
+
+    @swagger_auto_schema(
+        responses={200: CommunityPackageListSerializer()},
+        operation_id="experimental.frontend.community.packages",
+    )
+    def get(self, request: HttpRequest, **kwargs) -> HttpResponse:
+        communities = Community.objects.prefetch_related("package_categories")
+        community = get_object_or_404(communities, identifier=kwargs.get("community"))
```

Comment:
I'm not sure what you're actually suggesting here.

Reply:
TBH I see very little benefit from this regarding typing, since Django's URL dispatcher guarantees it to be a non-empty string, but I guess it can improve readability. I'll change the new experimental views to use this approach, although they will then work differently from the rest of the views (at least the ones I checked).

Reply:
This would be primarily for static checking for downstream use, as the variable is much cleaner/easier to type hint this way rather than type hinting the entire kwargs dict (which would be the alternative option). That being said, I'm not sure if you can even type hint the kwargs dict in a way where calling `.get` on it resolves correctly during static analysis.

Reply:
Assuming there's a downstream (`get_object_or_404`'s kwargs seem to be any-typed).